staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -475,20 +485,41 @@ func (p *applyPatcher) applyPatchToCurrentObject(_ context.Context, obj runtime.
 		panic("FieldManager must be installed to run apply")
 	}
 
+	klog.Warningf("p.patch: %v\n", string(p.patch))
+	// TODO: we need to strict unmarshal here
 	patchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
 	if err := yaml.Unmarshal(p.patch, &patchObj.Object); err != nil {
 		return nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
 	}
 
+	if p.validationDirective == metav1.FieldValidationStrict {
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+		if err := yaml.UnmarshalStrict(p.patch, &strictPatchObj.Object); err != nil {
+			return nil, errors.NewBadRequest(fmt.Sprintf("error strict decoding YAML: %v", err))
+		}
+
+	}
+	if p.validationDirective == metav1.FieldValidationWarn {
+		klog.Warningf("warn directive")
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
+		if err := yaml.UnmarshalStrict(p.patch, &strictPatchObj.Object); err != nil {
+			klog.Warningf("warn strict err: %v\n on context: %v\n", err, requestContext)
```

Comment:
I don't know if I would log a warning, the apiserver doesn't really have a warning, it's working perfectly fine :-)